connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndITSuite.scala

```diff
@@ -134,26 +137,127 @@ class SparkBigQueryEndToEndITSuite extends FunSuite
     }
   }
 
-  test("out of order columns") {
-    val row = spark.read.format("bigquery").option("table", SHAKESPEARE_TABLE).load()
-      .select("word_count", "word").head
-    assert(row(0).isInstanceOf[Long])
-    assert(row(1).isInstanceOf[String])
-  }
+  def testsWithReadInFormat(dataFormat: String): Unit = {
 
-  test("number of partitions") {
-    val df = spark.read.format("com.google.cloud.spark.bigquery")
-      .option("table", LARGE_TABLE)
-      .option("parallelism", "5")
-      .load()
-    assert(5 == df.rdd.getNumPartitions)
-  }
+    test("out of order columns. Data Format %s".format(dataFormat)) {
+      val row = spark.read.format("bigquery")
+        .option("table", SHAKESPEARE_TABLE)
+        .option("readDataFormat", dataFormat).load()
+        .select("word_count", "word").head
+      assert(row(0).isInstanceOf[Long])
+      assert(row(1).isInstanceOf[String])
+    }
 
-  test("default number of partitions") {
-    val df = spark.read.format("com.google.cloud.spark.bigquery")
-      .option("table", LARGE_TABLE)
-      .load()
-    assert(df.rdd.getNumPartitions == 35)
+    test("number of partitions. Data Format %s".format(dataFormat)) {
+      val df = spark.read.format("com.google.cloud.spark.bigquery")
+        .option("table", LARGE_TABLE)
+        .option("parallelism", "5")
+        .option("readDataFormat", dataFormat)
+        .load()
+      assert(5 == df.rdd.getNumPartitions)
+    }
+
+    test("default number of partitions. Data Format %s".format(dataFormat)) {
+      val df = spark.read.format("com.google.cloud.spark.bigquery")
+        .option("table", LARGE_TABLE)
+        .option("readDataFormat", dataFormat)
+        .load()
+      assert(df.rdd.getNumPartitions == 35)
```

Comment:
why this is 35 isn't clear to me.

Reply:
35 is default number of partitions so this test verifies that the output follows it. 

@davidrabinowitz has more context on this.